### PR TITLE
feat: purchase order against sales order

### PR DESCRIPTION
**Pull request by Anurag810**
taskbyrohit

### CODE REVIEW

**Comment by Anurag810** [comment]
Removed add row and delete buttons from table field.
Updated screenshot for that.
Screenshot 2019-06-18 at 4 18 32 PM

File: erpnext/selling/doctype/sales_order/sales_order.js [hunk 1]
```diff
@@ -129,28 +129,11 @@ erpnext.selling.SalesOrderController = erpnext.selling.SellingController.extend(
...
-							if(item.qty > flt(item.delivered_qty)) {
-								allow_delivery = true;
-							}
-						}
 
-						if (allow_delivery && allow_purchase) {
-							break;
-						}
-					}
+					allow_purchase = true
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_order/sales_order.js]: remove allow_purchase

File: erpnext/selling/doctype/sales_order/sales_order.py [hunk 2]
```diff
@@ -765,7 +765,10 @@ def get_events(start, end, filters=None):
 	return data
 
 @frappe.whitelist()
-def make_purchase_order_for_drop_shipment(source_name, for_supplier=None, target_doc=None):
+def make_purchase_order_for_drop_shipment(source_name, for_supplier=None, selected_items=[], target_doc=None):
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_order/sales_order.py]: As now PO can be made against any item in SO, remove the drop_shipment term from function naming

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/selling/doctype/sales_order/sales_order.js (hunk 1 above)
Comment (Anurag810) [reply erpnext/selling/doctype/sales_order/sales_order.js]: done

File: erpnext/selling/doctype/sales_order/sales_order.py (hunk 2 above)
Comment (Anurag810) [reply erpnext/selling/doctype/sales_order/sales_order.py]: done

**Review by nabinhait — APPROVED** [review]

**Pull request merged**